zrml/prediction-markets/src/migrations.rs

```diff
@@ -424,10 +420,289 @@ mod tests {
+
+            for id in bounded_vec.into_inner().iter() {
+                if let Ok(market) = <zrml_market_commons::Pallet<T>>::market(id) {
+                    if market.dispute_mechanism == MarketDisputeMechanism::Authorized {
+                        authorized_resolutions.insert(*id, resolve_at);
+                    }
+                } else {
+                    log::warn!("AddFieldToAuthorityReport: Could not find market with id {:?}", id);
+                }
+            }
+        }
+
+        let mut new_storage_map: Vec<(
+            <T as zrml_market_commons::Config>::MarketId,
+            AuthorityReport<BlockNumberFor<T>>,
+        )> = Vec::new();
+
+        let now = frame_system::Pallet::<T>::block_number();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+        for (market_id, old_value) in storage_key_iter::<
+            <T as zrml_market_commons::Config>::MarketId,
+            OutcomeReport,
+            Twox64Concat,
+        >(AUTHORIZED, AUTHORIZED_OUTCOME_REPORTS)
+        {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+            let resolve_at: Option<BlockNumberFor<T>> =
+                authorized_resolutions.get(&market_id).cloned();
+
+            match resolve_at {
+                Some(block) if now <= block => {
+                    new_storage_map.push((
+                        market_id,
+                        AuthorityReport { resolve_at: block, outcome: old_value },
+                    ));
+                }
+                _ => {
+                    log::warn!(
+                        "AddFieldToAuthorityReport: Market was not found in \
+                         MarketIdsPerDisputeBlock; market id: {:?}",
+                        market_id
+                    );
+                    // example case market id 432
+                    // https://github.com/zeitgeistpm/zeitgeist/pull/701 market id 432 is invalid, because of zero-division error in the past
+                    // we have to handle manually here, because MarketIdsPerDisputeBlock does not contain 432
+                    let mut resolve_at = now.saturating_add(T::CorrectionPeriod::get());
+                    total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+                    let mut bounded_vec = <crate::MarketIdsPerDisputeBlock<T>>::get(resolve_at);
+                    while bounded_vec.is_full() {
+                        // roll the dice until we find a block that is not full
+                        total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+                        resolve_at = resolve_at.saturating_add(1u32.into());
+                        bounded_vec = <crate::MarketIdsPerDisputeBlock<T>>::get(resolve_at);
+                    }
+                    // is not full, so we can push
+                    bounded_vec.force_push(market_id);
```

Comment:
why force_push() is required?

Reply:
It is not directly required, I just used it. You can also use `try_push` with an unwrap. It is just another flavour of code. I check if the bounded vector is full above, if not, then force push.